frontend/app/src/util/AppNavigation.ts

```diff
@@ -291,22 +305,36 @@ export class AppNavigation {
 
   readonly onPageNotFound: PageNotFoundCallback
 
+  readonly onPageIconChange: SetIconCallback
+
+  isPageTitleSet: boolean
+
+  isPageIconSet: boolean
+
   strategy: StrategyV1 | StrategyV2
 
   constructor(
     hostCommunicationMgr: HostCommunicationManager,
     onUpdatePageUrl: PageUrlUpdateCallback,
-    onPageNotFound: PageNotFoundCallback
+    onPageNotFound: PageNotFoundCallback,
+    onPageIconChange: SetIconCallback
   ) {
     this.hostCommunicationMgr = hostCommunicationMgr
     this.onUpdatePageUrl = onUpdatePageUrl
     this.onPageNotFound = onPageNotFound
+    this.onPageIconChange = onPageIconChange
+    this.isPageIconSet = false
+    this.isPageTitleSet = false
 
     // Start with the V1 strategy as it will apply to V0 as well
     this.strategy = new StrategyV1(this)
   }
 
   handleNewSession(newSession: NewSession): MaybeStateUpdate {
+    // Reset the page title and icon
```

Comment:
nit: i feel like it's clear enough from the code what's going on here that the comment doesn't add anything / can be removed